src/common/src/types/jsonb.rs

```diff
@@ -478,3 +416,100 @@ impl<F: std::fmt::Write> std::io::Write for FmtToIoUnchecked<F> {
         Ok(())
     }
 }
+impl From<F32> for Value {
+    fn from(v: F32) -> Self {
+        if v.0 == f32::INFINITY {
+            "Infinity".into()
+        } else if v.0 == f32::NEG_INFINITY {
+            "-Infinity".into()
+        } else if v.0.is_nan() {
+            "NaN".into()
+        } else {
+            v.0.into()
+        }
+    }
+}
+
+// NOTE: Infinite or NaN values are not JSON numbers. They are stored as strings in Postgres.
+impl From<F64> for Value {
+    fn from(v: F64) -> Self {
+        if v.0 == f64::INFINITY {
+            "Infinity".into()
+        } else if v.0 == f64::NEG_INFINITY {
+            "-Infinity".into()
+        } else if v.0.is_nan() {
+            "NaN".into()
+        } else {
+            v.0.into()
+        }
+    }
+}
+
+impl From<JsonbRef<'_>> for Value {
+    fn from(v: JsonbRef<'_>) -> Self {
+        v.0.to_owned()
+    }
+}
+
+impl From<Decimal> for Value {
+    fn from(v: Decimal) -> Self {
+        // Make it numeric if it can be converted to IEEE 754 double, otherwise a string.
+        // May loss precision here.
+        let res: Result<F64, _> = v.try_into();
+        if let Ok(value) = res {
+            value.into()
+        } else {
+            v.to_text().as_str().into()
+        }
+    }
+}
+
+impl From<Timestamp> for Value {
+    fn from(v: Timestamp) -> Self {
+        let mut builder = Builder::default();
+        builder.display(format!("{}T{}", v.0.date(), v.0.time()));
```

Comment:
to truly avoid String allocation:
```suggestion
        builder.display(format_args!("{}T{}", v.0.date(), v.0.time()));
```